### DISCUSSION TITLE
Any advice on switching from read-only gallery to external library?

### DISCUSSION THREAD

**Discussion by davidecavestro** [immich-app/immich#4250] [Q&A]
I started importing photos with [read-only gallery](https://documentation.immich.app/docs/features/read-only-gallery), then upgraded Immich and switched to [external library](https://documentation.immich.app/docs/features/libraries#external-libraries): I created an external library having the same import path as I originally set for the CLI import and did a rescan.

Now I suppose all assets have been duplicated
![image](https://github.com/immich-app/immich/assets/447917/6997db15-755f-4b69-8237-38432c5d15dd)
in fact if I run this query
```sql
SELECT
  COUNT(*), string_agg("id"::text, ', '), "originalPath"
FROM
  assets
GROUP BY
  "originalPath"
HAVING
  COUNT(*) > 1
ORDER BY
  COUNT(*) DESC
LIMIT 5;
```
I see a lot of assets sharing the same import path
![image](https://github.com/immich-app/immich/assets/447917/616cde23-5a05-4208-9fb3-0d25c16dfd96)

So the question is: what's the best path to switch from readonly galleries to external libraries?

**Comment by maltokyo** [comment]
I have the same question..  but there is a duplicate exact same discussion - maybe we could choose just one?
I have 115,000 photos already fully indexed etc, and dont want to do that process again, as it took 3 days to process everything in the "external gallery"?

https://github.com/immich-app/immich/discussions/4240

**Comment by davidecavestro** [comment]
Sorry, this duplicates https://github.com/immich-app/immich/discussions/4240
